backend/stats/views.py

```diff
@@ -7,49 +7,51 @@
 from rest_framework import status
 from django.db.models import Count, Q
 from User.models import User
+from .user_stats_table import UserTable
+from .leader_board_table import LeaderBoardTable
+from .friend_stats_table import FirendsTable
 
+from .mixins import UserTableMixin
+from rest_framework.response import Response
 
-class StatsView(APIView):
+class StatsView(APIView, UserTableMixin):	
 	def get(self, request):
-		x = [1, 2, 3, 4, 5]
-		y = [10, 11, 12, 13, 14]
-		fig = go.Figure(data=[go.Table(header=dict(values=['A Scores', 'B Scores']),
-                 cells=dict(values=[[100, 90, 80, 90], [95, 85, 75, 95]]))
-                     ])
-
-        # Convert plot to HTML
-		plot_html = pio.to_html(fig, full_html=True)
+		user = request.user
+		wins = self.get_wins_count(user)
+		losses = self.get_losses_count(user)
+		total_game_played = self.get_total_games_played_count(user)
+		win_loss_ratio = self.get_win_loss_ratio(user)
+		wining_streak = self.get_win_streak_count(user)
+		position_in_leaderboard = self.get_position_in_leaderboard(user)
+		self.user_table = UserTable(user, wins, losses, total_game_played, win_loss_ratio, wining_streak, position_in_leaderboard)
+        # Create the Plotly figure
+		fig = go.Figure(data=[go.Table(
+			header=dict(values=['Statistic', 'Value']),
+			cells=dict(values=[
+				['Wins', 'Losses', 'Total Games Played', 'Win/Loss Ratio', 'Winning Streak', 'Position in Leaderboard'],
+				[wins, losses, total_game_played, win_loss_ratio, wining_streak, position_in_leaderboard]
+			])
+		)])		
+	
+		# Convert the Plotly figure to an SVG string
+		svg_str = fig.to_image(format="svg").decode("utf-8")
 
+		return HttpResponse(svg_str, content_type='image/svg+xml')
 
-		return HttpResponse(plot_html, content_type='text/html')
-	
-class LeaderBoardView(APIView):
+class LeaderBoardView(APIView, UserTableMixin):
+	def __init__(self):
+		self.leader_table = LeaderBoardTable
+        
 	def get(self, request):
 		users_with_most_wins = User.objects.annotate(total_wins=Count('players', filter=Q(players__match_winner=True))).order_by('-total_wins')
```

Comment:
I don't think you should have an __init__ in a view. How about you put almost everything you have in the get method into a mixin that returns Response(json_response_data, status=status.HTTP_200_OK)?
Then you could just do:
def get(self, request):
    return self.leader_table_response()

Reply:
@sakarisalmi done, I moved the whole thing to mixing. Do you think I need to have a separate app for leaderboard as well. I personally think it's an overkill, but if you think it's necessary then sure.

Reply:
> Somebody discovered mixins xD

I have, and I'm in love 😍

Reply:
> Do you think I need to have a separate app for leaderboard as well. I personally think it's an overkill.

No, but you can always use apps inside of apps, so if it makes it easier for you, you might want to do something like that.